### PR TITLE
Added solution

### PR DESCRIPTION
**Pull request by Nokafan**
The solution for delete product from the cart is not ready, please don't check DeleteProductFromCartController. I don't know how to remove them from git control, so I live them for the moment.

### CODE REVIEW

File: src/main/java/com/internet/shop/controllers/DeleteProductFromCartController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.interfaces.ProductService;
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductFromCartController extends HttpServlet {
```
Comment (OleksiiTromsa) [inline src/main/java/com/internet/shop/controllers/DeleteProductFromCartController.java]: This is task from next homework, so don't worry :)

File: src/main/java/com/internet/shop/controllers/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.internet.shop.model.Product;
+import com.internet.shop.service.interfaces.ProductService;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ProductController extends HttpServlet {
```
Comment (OleksiiTromsa) [inline src/main/java/com/internet/shop/controllers/ProductController.java]: We have a point in our checklist:
"Use informative names for your controllers. Always start controller’s name with verb: GetAllUsersController, AddProductToShoppingCartController etc."

File: src/main/webapp/WEB-INF/views/products/addProduct.jsp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    <button type="submit">Add product</button>
+    <br><br>
+</form>
+<a href="${pageContext.request.contextPath}/">Main</a>
+<br><br>
+<a href="${pageContext.request.contextPath}/products/all">All products</a>
+<br><br>
+<a href="${pageContext.request.contextPath}/cart/all">Cart</a>
+</body>
+</html>
```
Comment (OleksiiTromsa) [inline src/main/webapp/WEB-INF/views/products/addProduct.jsp]: Empty line

File: src/main/webapp/WEB-INF/views/products/addProduct.jsp [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<head>
+    <title>Add product</title>
+</head>
+<body>
+<h1>Add product</h1>
+<h1>please add information</h1>
+<h2 style="color: blue">${message}</h2>
+<form method="post" action="${pageContext.request.contextPath}/products/addProduct">
+    Name: <input type="text" name="name">
+    Price: <input type="number" step="0.01" name="price">
```
Comment (OleksiiTromsa) [inline src/main/webapp/WEB-INF/views/products/addProduct.jsp]: According to checklist we should make here a check whether the field empty and use jsp tag for it.
Same for user registration.

File: src/main/webapp/WEB-INF/views/products/all.jsp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+<br><br>
+<a href="${pageContext.request.contextPath}/">To main page</a>
+<br><br>
+<a href="${pageContext.request.contextPath}/cart">Cart</a>
+</body>
+</html>
```
Comment (OleksiiTromsa) [inline src/main/webapp/WEB-INF/views/products/all.jsp]: empty line

File: src/main/webapp/WEB-INF/views/users/all.jsp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+                <c:out value="${user.login}"/>
+            </td>
+            <td>
+                <a href="${pageContext.request.contextPath}/users/delete?id=${user.id}">Delete</a>
+            </td>
+        </tr>
+    </c:forEach>
+</table>
+</body>
+</html>
```
Comment (OleksiiTromsa) [inline src/main/webapp/WEB-INF/views/users/all.jsp]: empty line

File: src/main/webapp/WEB-INF/views/products/addProduct.jsp (hunk 3 above)
Comment (Nokafan) [reply src/main/webapp/WEB-INF/views/products/addProduct.jsp]: Thx. Fixed.

File: src/main/webapp/WEB-INF/views/products/addProduct.jsp (hunk 4 above)
Comment (Nokafan) [reply src/main/webapp/WEB-INF/views/products/addProduct.jsp]: Thx. Fixed.

File: src/main/webapp/WEB-INF/views/products/all.jsp (hunk 5 above)
Comment (Nokafan) [reply src/main/webapp/WEB-INF/views/products/all.jsp]: Thx. Fixed.

File: src/main/webapp/web.xml [hunk 7]
```diff
@@ -4,13 +4,95 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>injectData</servlet-name>
+        <servlet-class>com.internet.shop.controllers.InjectDataController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>injectData</servlet-name>
+        <url-pattern>/injectData</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -4,13 +4,95 @@
...
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>addProducts</servlet-name>
+        <servlet-class>com.internet.shop.controllers.AddProductController</servlet-class>
+    </servlet>
+
+    <servlet-mapping>
+        <servlet-name>addProducts</servlet-name>
+        <url-pattern>/products/addProduct</url-pattern>
```
Comment (Sofasmile) [inline src/main/webapp/web.xml]: https://mate-academy.github.io/jv-program-fulltime/02_jdbc_and_servlets/checklist/hw_16_1.html

**Review by Sofasmile — COMMENTED** [review]
great job

File: src/main/webapp/WEB-INF/views/users/all.jsp (hunk 6 above)
Comment (Nokafan) [reply src/main/webapp/WEB-INF/views/users/all.jsp]: Thx. Fixed.

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**